shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Examples/PoolLifetime.hs

```diff
@@ -0,0 +1,207 @@
+    lastByronHeaderHash,
+    nonce0,
+    ppEx,
+  )
+import Test.Shelley.Spec.Ledger.Generator.Core
+  ( AllIssuerKeys (..),
+    NatNonce (..),
+    genesisCoins,
+    genesisId,
+    mkBlock,
+    mkOCert,
+    zero,
+  )
+import Test.Shelley.Spec.Ledger.Utils (getBlockNonce)
+
+aliceInitCoin :: Coin
+aliceInitCoin = 10 * 1000 * 1000 * 1000 * 1000 * 1000
+
+bobInitCoin :: Coin
+bobInitCoin = 1 * 1000 * 1000 * 1000 * 1000 * 1000
+
+initUTxO :: Crypto c => UTxO c
+initUTxO =
+  genesisCoins
+    [ TxOut Cast.aliceAddr aliceInitCoin,
+      TxOut Cast.bobAddr bobInitCoin
+    ]
+
+initStEx2 :: forall c. Crypto c => ChainState c
+initStEx2 = initSt initUTxO
+
+-- SET UP BLOCK
+
+aliceCoinEx2A :: Coin
+aliceCoinEx2A = aliceInitCoin - (_poolDeposit ppEx) - 3 * (_keyDeposit ppEx) - 3
+
+carlMIR :: Coin
+carlMIR = Coin 110
+
+dariaMIR :: Coin
+dariaMIR = Coin 99
+
+feeTx2A :: Coin
+feeTx2A = Coin 3
+
+txbodyEx2A :: Crypto c => TxBody c
+txbodyEx2A =
+  TxBody
+    (Set.fromList [TxIn genesisId 0])
+    (StrictSeq.fromList [TxOut Cast.aliceAddr aliceCoinEx2A])
+    ( StrictSeq.fromList
+        ( [ DCertDeleg (RegKey Cast.aliceSHK),
+            DCertDeleg (RegKey Cast.bobSHK),
+            DCertDeleg (RegKey Cast.carlSHK),
+            DCertPool (RegPool Cast.alicePoolParams)
+          ]
+            ++ [ DCertMir
+                   ( MIRCert
+                       ReservesMIR
+                       ( Map.fromList
+                           [ (Cast.carlSHK, carlMIR),
+                             (Cast.dariaSHK, dariaMIR)
+                           ]
+                       )
+                   )
+               ]
+        )
+    )
+    (Wdrl Map.empty)
+    feeTx2A
+    (SlotNo 10)
+    SNothing
+    SNothing
+
+txEx2A :: forall c. Mock c => Tx c
+txEx2A =
+  Tx
+    txbodyEx2A
+    mempty
+      { addrWits =
+          makeWitnessesVKey
+            (hashAnnotated txbodyEx2A)
+            ( (asWitness <$> [Cast.alicePay, Cast.carlPay])
+                <> (asWitness <$> [Cast.aliceStake])
+                <> [asWitness $ cold Cast.alicePoolKeys]
+                <> ( asWitness
+                       <$> [ cold (coreNodeIssuerKeys 0),
+                             cold (coreNodeIssuerKeys 1),
+                             cold (coreNodeIssuerKeys 2),
+                             cold (coreNodeIssuerKeys 3),
+                             cold (coreNodeIssuerKeys 4)
+                           ]
+                   )
+            )
+      }
+    SNothing
+
+blockEx2A :: forall c. (HasCallStack, Mock c) => Block c
+blockEx2A =
+  mkBlock
+    lastByronHeaderHash
+    (coreNodeKeysBySchedule ppEx 10)
+    [txEx2A]
+    (SlotNo 10)
+    (BlockNo 1)
+    (nonce0 @c)
+    (NatNonce 1)
+    zero
+    0
+    0
+    (mkOCert (coreNodeKeysBySchedule ppEx 10) 0 (KESPeriod 0))
+
+blockNonce2A :: forall c. (HasCallStack, Mock c) => Nonce
+blockNonce2A = getBlockNonce (blockEx2A @c)
+
+expectedStEx2A :: forall c. Mock c => ChainState c
+expectedStEx2A =
```

Comment:
This is a good example of how the new example will read. `expectedStEx2A` is obtained by applying the combinators to `initStEx2`.